templates/jenkins-slave-defaults.erb

```diff
@@ -54,9 +54,9 @@ TOOL_LOCATIONS="<%= @tool_locations -%>"
 
 DESCRIPTION="<%= @description -%>"
 
-# credentials
-JENKINS_USERNAME="<%= @ui_user -%>"
-JENKINS_PASSWORD="<%= @ui_pass -%>"
+# credentials should be single quoted
+JENKINS_USERNAME="'<%= @ui_user.gsub!(/^\'|\'?$/, '') -%>'"
```

Comment:
This will raise an exception when `@ui_user` is `nil` (`undef`). 

Why isn't `JENKINS_USERNAME='<%= @ui_user -%>'` sufficient to stop the shell from interpolating the string?